lib/ddtrace/configuration/settings.rb

```diff
@@ -483,14 +499,14 @@ def initialize(*_)
         # TODO: This setting is not tracer-specific and should be moved to top-level or to the transport.
         option :hostname
```

Comment:
@marcotc and I thought about this. Main reason we didn't yet was it was unclear where these should go. Should they be top-level? Or should they be grouped under `writer`/`api`/`agent`? Should the global settings define connectivity endpoints and the tracer just use them?

There was a lot of directions to go with this, but we didn't feel like we had the time to adequately assess this one.

Reply:
Looks like @marcotc had started [a PR](https://github.com/DataDog/dd-trace-rb/pull/1817) to move this to `agent.hostname` which sounds fine to me.

One thing we should keep in mind is tracing has its own unique `port`, and `hostname` theoretically could be too (if metrics/profiles were sent to one agent, and traces were sent to another.) This should probably be preserved as a trace specific setting.

Reply:
> One thing we should keep in mind is tracing has its own unique port

Could you expand a bit on this one? I thought we basically had two ports going on:

a) send stats via udp and
b) talk to the agent via tcp (for sending traces, sending profiles, and possibly more in the future)

So it sounds like I'm missing something?

(On the topic of changing this, you have a point that this needs a bigger change -- especially `transport_options`, so yeah maybe leave for post-1.0 if we don't have time for it now.)

Reply:
A user could want to configure two agents: one to accept profiles, another to accept traces. Presumably this might be done if its a high volume application and splitting the load would be of a benefit.

Reply:
I'm not entirely convinced... The load from profiling is really tiny -- the agent effectively acts as a dumb proxy. Has this use case been requested by customers? Do other Datadog libraries provide this already?

Reply:
(While I'm pushing back a bit on this use case, I can understand if we decide to leave it as-is for now for time/priority constraints -- I'm just not convinced that there's a reason for this structure other than an historical one)